### ISSUE TITLE
[QUESTION] jsPanel tooltip - how to destroy it?

### ISSUE THREAD

**Issue by temuri416** [Flyer53/jsPanel4#91]
Hello,

My tooltip is created as:

```js
jsPanel.tooltip.create({
    header: title,
    headerTitle: title,
    content: message,
    target: el,
    delay: 0,
    boxShadow: 1,
    connector: '#dadada',
    border: '1px solid darkgray',
    position: {
        my: 'center-top',
        at: 'center-bottom',
    },
    contentSize: 'auto',
    theme: '#eeeeee filledlight'
});
```

I need to be able to re-attach a different tooltip to `el` element on the fly. How do I unset previously defined tooltip?

Thanks!

**Comment by Flyer53** [comment]
I‘ll take a look at that next week. I‘m away over the weekend ...

**Comment by temuri416** [comment]
May I suggest that you use target element data to store the tooltip instance?

**Comment by Flyer53** [comment]
That won't help because the actual tooltip is re-created each time the mouse enters the target and closed when mouse leaves the target.
You'll need to remove the event listener from the target (a specific handler because a target can have several tooltips as you know). I have an idea and might have something for you to test tomorrow ...
It probably will be an additional method `jsPanel.tooltip.remove(args);`

**Comment by Flyer53** [comment]
For testing I added the above mentioned method.
[Download modified version](https://alpha.jspanel.de/downloads/jspanel-4.7.0-beta.1-tooltiptest.zip) (should be ok to swap only the tooltip extension) and then modify your code example from above like:
```javascript
jsPanel.tooltip.create({
    header: title,
    headerTitle: title,
    content: message,
    target: el,  // will be used in new method 
    delay: 0,
    ttipName: 'tooltip-1',  // new option needed to identify the specific tooltip
    boxShadow: 1,
    connector: '#dadada',
    border: '1px solid darkgray',
    position: {
        my: 'center-top',
        at: 'center-bottom',
    },
    contentSize: 'auto',
    theme: '#eeeeee filledlight'
});
```
To remove the tooltip call:
```javascript
jsPanel.tooltip.remove(el, 'mouseenter', 'tooltip-1');
// 1st arg: the target
// 2nd arg: the event used to trigger the tooltip
// 3rd arg: the name of the tooltip (new option ttipName)
```
Check it out and let me know how it works out please.

Note: If you don't change the default `ttipEvent` and never have more than one tooltip attached to a specific target you can call `jsPanel.tooltip.remove()` with the target as only argument.

**Comment by temuri416** [comment]
Thanks, I'm currently testing that.

FYI, updating entire jsPanel from 4.6.0 to 4.7.0-beta.1 broke tooltip sizing when content padding is applied to tooltip header and content area:

```css
.jsPanel.myTooltip .jsPanel-hdr .jsPanel-headerbar,
.jsPanel.myTooltip .jsPanel-content {padding:5px 15px}
```

In 4.6.0 overflow control wasn't showing.

![image](https://user-images.githubusercontent.com/3987203/58805659-80b59e00-85e2-11e9-9489-b69d21d03c6e.png)

**Comment by temuri416** [comment]
Tested tooltip removal - it worked fine.

I'm wondering whether it's possible to auto-generate `ttipName` parameter automatically (if omitted) - based on DOM element's properties?

**Comment by Flyer53** [comment]
**Scrollbar:**
I have yet to investigate where that comes from. Behaviour is inconsistent across browser. Chrome for example shows the scrollbar, FF does not. At least in my dev environment. I gues you're using `auto` for the height. Then it's probably just a few pixels. For now you could set option `contentOverflow` to `'hidden'`.

**ttipName:**
It actually defaults to ... well ... `'default'`. You need to set `ttipName` only if you will have more than one tooltip at the same time on a given target.
`ttipEvent` defaults to `'mouseenter'`.
So if you have only one tooltip on a target and did not change `ttipEvent` and `ttipName` you can remove the tooltip with `jsPanel.tooltip.remove(target);`
If you need to auto-generate ttipName you could use an IIFE like
```javascript
jsPanel.tooltip.create({
  // ...
  ttipName: (function() { /* return a string to use */ })(),
  // ...
});
```
Did not try that but it should work.

**Comment by Flyer53** [comment]
New tooltip method to remove a tooltip handler implemented in v4.7.0
